src/Arcus.Messaging.ServiceBus.Core/ServiceBusMessageBuilder.cs

```diff
@@ -57,7 +57,23 @@ public static ServiceBusMessageBuilder CreateForBody(object messageBody, Encodin
         public ServiceBusMessageBuilder WithOperationId(string operationId)
         {
             Guard.NotNullOrWhitespace(operationId, nameof(operationId), "Requires a non-blank operation ID for the Azure Service Bus message");
-            _operationId = operationId;
+
+            _operationIdProperty = new KeyValuePair<string, object>(null, operationId);
+            return this;
+        }
+
+        /// <summary>
+        /// Adds an <paramref name="operationId"/> as an application property in the <see cref="ServiceBusMessage.ApplicationProperties"/>.
+        /// </summary>
+        /// <param name="operationId">The unique identifier for this operation.</param>
+        /// <param name="operationIdPropertyName">The name of the application property of the <paramref name="operationIdPropertyName"/>.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="operationId"/> or the <paramref name="operationIdPropertyName"/> is blank.</exception>
+        public ServiceBusMessageBuilder WithOperationId(string operationId, string operationIdPropertyName)
```

Comment:
What are we doing with this `operationPropertyName`  ? What is the purpose ?  That is not very clear to me.  The explanation in the param-description (xml docs) doesn't help too much either :)

Reply:
The `operationIdPropertyName` is the name of the application property where the `operationId` will be placed. Like the summary explains, this will be placed in the `ServceBusMessage.ApplicationProperties` property. This is a dictionary, so the overload here helps with providing a custom property name for the operation name.
What could help with this explanation?

Reply:
Updated the code docs: https://github.com/arcus-azure/arcus.messaging/pull/282/commits/faf90bc50744cbf6abe6d364a643c14936a647ba

Reply:
I thought it would be something like that :)
I did a small suggestion.

Reply:
😉 , ok, great!